hw1_classes/code/task5.py

```diff
@@ -0,0 +1,122 @@
+    def __getitem__(self):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+
+class AcidSequence(BiologicalSequence):
+    def __init__(self, sequence):
+        self._sequence = sequence
+        self._DICT_COMPLEMENT = None
+
+    def __len__(self):
+        return len(self._sequence)
+
+    def __getitem__(self, idx):
+        return self._sequence.__getitem__(idx)
+
+    def __str__(self):
+        return str(self._sequence)
+
+    def check_alphabet(self):
+        if self._DICT_COMPLEMENT is None:
+            raise NotImplementedError(
+                'Define check_alphabet in one of the following classes: %s.' % (type(self).__subclasses__()))
+        else:
+            for nucl in self._sequence:
+                if nucl.upper() not in self._DICT_COMPLEMENT:
+                    return False
+            return True
+
+
+class NucleicAcidSequence(AcidSequence):
+    def __init__(self, sequence):
+        super().__init__(sequence)
+
+    def complement(self):
+        if self._DICT_COMPLEMENT is None:
+            raise NotImplementedError(
+                'Define complement in one of the following classes: %s.' % (type(self).__subclasses__()))
+        else:
+            complement_seq = []
+            for nucl in self._sequence:
+                complement_seq.append(self._DICT_COMPLEMENT[nucl])
+            return ''.join(complement_seq[::-1])
+
+    def gc_content(self):
+        gc_compound = 0
+        total_len = len(self._sequence)
+        for nucl in self._sequence:
+            if nucl.upper() == 'G' or nucl.upper() == 'C':
+                gc_compound += 1
+        return gc_compound / total_len * 100
+
+
+class AminoAcidSequence(AcidSequence):
+    def __init__(self, sequence):
+        super().__init__(sequence)
+        self._DICT_COMPLEMENT = {'A': 'Ala',
+                                 'R': 'Arg',
+                                 'N': 'Asn',
+                                 'D': 'Asd',
+                                 'C': 'Cys',
+                                 'Q': 'Gln',
+                                 'E': 'Glu',
+                                 'G': 'Gly',
+                                 'H': 'His',
+                                 'I': 'Iso',
+                                 'L': 'Leu',
+                                 'K': 'Lys',
+                                 'M': 'Met',
+                                 'F': 'Phe',
+                                 'P': 'Pro',
+                                 'S': 'Serine',
+                                 'T': 'Thr',
+                                 'W': 'Trp',
+                                 'Y': 'Tyr',
+                                 'V': 'Val',
+                                 'X': 'Glx',
+                                 'Z': 'Gli',
+                                 'J': 'Nle',
+                                 'U': 'Cyc'}
+
+    def translate(self):
```

Comment:
Нейминг! Это не трансляция, а запись в трёхбуквенном формате
```suggestion
    def three_letter_format(self):
```